core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/scan/CSVScanSourceOpDesc.scala

```diff
@@ -0,0 +1,108 @@
+import edu.uci.ics.texera.workflow.common.metadata.{OperatorGroupConstants, OperatorInfo}
+import edu.uci.ics.texera.workflow.common.operators.source.SourceOperatorDescriptor
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType, Schema}
+import org.codehaus.jackson.map.annotate.JsonDeserialize
+
+import java.io.{File, IOException}
+import java.nio.charset.Charset
+import scala.collection.JavaConverters._
+class CSVScanSourceOpDesc extends SourceOperatorDescriptor {
+
+  @JsonProperty(required = true)
+  @JsonSchemaTitle("File")
+  @JsonDeserialize(contentAs = classOf[java.lang.String])
+  var fileName: Option[String] = None
+
+  @JsonProperty(defaultValue = ",")
+  @JsonSchemaTitle("Delimiter")
+  @JsonPropertyDescription("delimiter to separate each line into fields")
+  @JsonDeserialize(contentAs = classOf[java.lang.String])
+  var delimiter: Option[String] = None
+
+  @JsonProperty(defaultValue = "true")
+  @JsonSchemaTitle("Header")
+  @JsonPropertyDescription("whether the CSV file contains a header line")
+  var hasHeader: Boolean = true
+
+  @throws[IOException]
+  override def operatorExecutor: CSVScanSourceOpExecConfig = {
+    // fill in default values
+    if (delimiter.get.isEmpty)
+      delimiter = Option(",")
+
+    fileName match {
+      case Some(path) =>
+        val headerLine =
+          Files.asCharSource(new File(path), Charset.defaultCharset).readFirstLine
+
+        new CSVScanSourceOpExecConfig(
+          this.operatorIdentifier,
+          Constants.defaultNumWorkers,
+          path,
+          delimiter.get.charAt(0),
+          this.inferSchema(headerLine),
+          hasHeader
+        )
+      case None =>
+        throw new RuntimeException("File path is not provided.")
+    }
+
+  }
+
+  private def inferSchema(headerLine: String): Schema = {
+    if (delimiter.isEmpty) return null
+
+    val headers: Array[String] = headerLine.split(delimiter.get)
+    Schema.newBuilder
+      .add(
+        if (hasHeader)
+          headers
+            .map((c: String) => { c.trim })
+            .map((c: String) => new Attribute(c, AttributeType.STRING))
+            .toIterable
+            .asJava
+        else
+          headers.indices
+            .map((i: Int) => new Attribute("column" + i, AttributeType.STRING))
+            .asJava
+      )
+      .build
+
+  }
+
+  override def operatorInfo: OperatorInfo = {
+    OperatorInfo(
+      "CSV File Scan",
+      "Scan data from a local CSV file",
+      OperatorGroupConstants.SOURCE_GROUP,
+      0,
+      1
+    )
+  }
+
+  @throws[IOException]
+  override def sourceSchema(): Schema = {
+    if (fileName.isEmpty) return null
+
+    val headerLine =
+      Files.asCharSource(new File(fileName.get), Charset.defaultCharset).readFirstLine
+    inferSchema(headerLine)
+
+  }
+
+  override def setContext(workflowContext: WorkflowContext): Unit = {
+    super.setContext(workflowContext)
+    if (context.userID.isDefined)
+      fileName = Option(
```

Comment:
I prefer to not override the setContext function and replace file name with file path here. Let's store file path as a separate variable and convert file name to file path when its needed.

Reply:
As I continue looking the changes, I now understand the reason you made it into one variable because you want to make it work without userID.  It seems a bit hacky to me but sadly I cannot think of any better solution to do it.  

Can we at least add a comment to say it should be a file path (when no user id), and file name (when user id exists, and explain the reason that we do this?

Reply:
added filePath and comments, can you check?